### PR TITLE
Refine setTexture

### PR DESCRIPTION
**Pull request by fusijie**

### CODE REVIEW

**Comment by fusijie** [comment]
To fix the first issue.

https://github.com/cocos2d/cocos2d-x/issues/10086

**Comment by pandamicro** [comment]
I think the API changes for getTextureForKey is irrelevant with the bug fix, it should probably be proposed in another PR

File: cocos/renderer/CCTextureCache.h [hunk 1]
```diff
@@ -146,9 +146,10 @@ class CC_DLL TextureCache : public Ref
 
     /** Returns an already created texture. Returns nil if the texture doesn't exist.
     @param key It's the related/absolute path of the file image.
+    @param checkFullPath If key not found, check key = fullPathForFilename(key). Default true
     @since v0.99.5
     */
-    Texture2D* getTextureForKey(const std::string& key) const;
+    Texture2D* getTextureForKey(const std::string& key, bool checkFullPath = true) const;
```
Comment (zilongshanren) [inline cocos/renderer/CCTextureCache.h:152]: default arguments are forbid in Cocos Style Guiding.

File: cocos/2d/CCSprite.cpp [hunk 2]
```diff
@@ -343,14 +343,14 @@ void Sprite::setTexture(const std::string &filename)
...
+    CCASSERT(!_batchNode || !texture || texture->getName() == _batchNode->getTexture()->getName(), "CCSprite: Batched sprites should use the same texture as the batchnode");
     // accept texture==nil as argument
-    CCASSERT( !texture || dynamic_cast<Texture2D*>(texture), "setTexture expects a Texture2D. Invalid argument");
+    CCASSERT(dynamic_cast<Texture2D*>(texture), "setTexture expects a Texture2D. Invalid argument");
 
     if (texture == nullptr)
     {
-        // Gets the texture by key firstly.
-        texture = Director::getInstance()->getTextureCache()->getTextureForKey(CC_2x2_WHITE_IMAGE_KEY);
+        // Gets the texture by key firstly. No check full path
```
Comment (zilongshanren) [inline cocos/2d/CCSprite.cpp:352]: Why not check the full path?

**Pull request closed without merging**